From 3fe79c1e6598702e0ae2b007a5ff10e53b8e1270 Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Mon, 22 Aug 2022 21:45:34 +0000
Subject: [PATCH] [RESTRICT AUTOMERGE] Restrict halfsheet cancel broadcast

Bug: 234442700
Test: atest: NearbyUnitTest
Tag: #security
Ignore-AOSP-First: Security
Change-Id: I7d7606e40928cb6cd2ed9c022b2375012baec1d6
(cherry picked from commit be99ea27e22ef97aec9a95ba6d5f52e954e9f2a7)
Merged-In: I7d7606e40928cb6cd2ed9c022b2375012baec1d6
---
 .../com/android/nearby/halfsheet/HalfSheetActivity.java    | 5 ++++-
 .../com/android/nearby/halfsheet/utils/BroadcastUtils.java | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/nearby/halfsheet/src/com/android/nearby/halfsheet/HalfSheetActivity.java b/nearby/halfsheet/src/com/android/nearby/halfsheet/HalfSheetActivity.java
index 2a38b8ad9..07e577637 100644
--- a/nearby/halfsheet/src/com/android/nearby/halfsheet/HalfSheetActivity.java
+++ b/nearby/halfsheet/src/com/android/nearby/halfsheet/HalfSheetActivity.java
@@ -16,6 +16,8 @@
 
 package com.android.nearby.halfsheet;
 
+import static android.Manifest.permission.ACCESS_FINE_LOCATION;
+
 import static com.android.nearby.halfsheet.fragment.DevicePairingFragment.APP_LAUNCH_FRAGMENT_TYPE;
 import static com.android.server.nearby.common.bluetooth.fastpair.FastPairConstants.EXTRA_MODEL_ID;
 import static com.android.server.nearby.common.fastpair.service.UserActionHandlerBase.EXTRA_MAC_ADDRESS;
@@ -226,7 +228,8 @@ public class HalfSheetActivity extends FragmentActivity {
                                     EXTRA_HALF_SHEET_IS_RETROACTIVE,
                                     getIntent().getBooleanExtra(EXTRA_HALF_SHEET_IS_RETROACTIVE,
                                             false))
-                            .putExtra(EXTRA_MAC_ADDRESS, mScanFastPairStoreItem.getAddress()));
+                            .putExtra(EXTRA_MAC_ADDRESS, mScanFastPairStoreItem.getAddress()),
+                    ACCESS_FINE_LOCATION);
         }
     }
 
diff --git a/nearby/halfsheet/src/com/android/nearby/halfsheet/utils/BroadcastUtils.java b/nearby/halfsheet/src/com/android/nearby/halfsheet/utils/BroadcastUtils.java
index 467997c06..2f1e90ae6 100644
--- a/nearby/halfsheet/src/com/android/nearby/halfsheet/utils/BroadcastUtils.java
+++ b/nearby/halfsheet/src/com/android/nearby/halfsheet/utils/BroadcastUtils.java
@@ -31,6 +31,13 @@ public class BroadcastUtils {
         context.sendBroadcast(intent);
     }
 
+    /**
+     * Helps send a broadcast with specified receiver permission.
+     */
+    public static void sendBroadcast(Context context, Intent intent, String receiverPermission) {
+        context.sendBroadcast(intent, receiverPermission);
+    }
+
     private BroadcastUtils() {
     }
 }
-- 
2.39.0.rc1.256.g54fd8350bd-goog

